### PR TITLE
Issue4

### PR DESCRIPTION
**Pull request by nomisto**
Changes:

- Added Identifiers (CURIE)
    - Identifiers are specified in the Infile classes with a Namespace object
    - Namespace objects contain the Namespace, a boolean if the namespace is already in the ID (f.e. GO, KEGG,...) and a dictionary for replacement (f.e. the identifier of the ID REACT: is actually REACTOME:)
    - Identifiers are added, when the graph is written to the output files
- Added the option to write the graph as N3
    - There are now 2 writer classes: GraphRDFWriter and GraphTSVWriter
- GUI
    - Format output can now be selected (TSV or RDF-N3)
    - Main is now started as a Daemon to keep the UI responsive
    - Progress is shown in the Console output
- README
    - Supplemented the source database table with the Identifiers of an edge, mapping, onto
    - Will maybe improve understanding of custom graph creation
- Suppression of pandas SettingWithCopyWarning

### CODE REVIEW

**Comment by matthias-samwald** [comment]
Issue 4 also contains the TODO "Add data source as fifth column in TSV file" -- is this also covered by this pull request?

**Comment by nomisto** [comment]
Oh no sorry, forgot about that, fixing it asap.

File: src/openbiolink/edge.py [hunk 1]
```diff
@@ -2,12 +2,13 @@
 
 
 class Edge:
-    def __init__(self, id1: str, type : EdgeType, id2: str, source: "", qScore = None):
```
Comment (nomisto) [inline src/openbiolink/edge.py]: I added a new parameter to the edge class called sourcedb. I noticed that there is already a parameter called source, i couldn't find where it is used, but didn't want to override it.

File: src/openbiolink/edge.py [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
+    def __init__(self, id1: str, type : EdgeType, id2: str, source: "", qScore = None, sourcedb =None):
```
Comment (nomisto) [inline src/openbiolink/edge.py]: I added a new parameter to the edge class called sourcedb. I noticed that there is already a parameter called source, i couldn't find where it is used, but didn't want to override it.

**Comment by matthias-samwald** [comment]
With the new code, can we regenerate the pre-existing train-test split or will the newly generated files have different triples for each split? (I guess the latter option, since there's a non-deterministic randomization split being done?)

**Comment by matthias-samwald** [comment]
@annabreit I don't have any further comments, do you?

(Perhaps we should also generate an official validation split in the next iteration of the benchmark that we will generate (e.g. a 90-5-5 train-validation-test split?). Anyways, this is probably off-topic for this pull request)

**Comment by nomisto** [comment]
There is currently also a bug when performing the train/test split with an edges file with no quality score.

As the headers of the dataframe are 
`[NODE1_ID_COL_NAME, EDGE_TYPE_COL_NAME, NODE2_ID_COL_NAME, QSCORE_COL_NAME, SOURCE_COL_NAME]`

and a edge file with no quality score would have the columns 
`Node1_ID, Edge_type, Node2_ID, Source`

The easiest fix to this problem would be in my opinion to swap the quality score and the db-source columns, would this be ok?

**Comment by matthias-samwald** [comment]
@nomisto Could you please generate new graph and benchmark files with the new code so we could review the output?
I suggest using a 90-5-5 train-validation-test split this time.
If there are no issues with the newly generated graphs/benchmarks and there are no complaints by Anna, we can pull the changes to the main branch.

**Comment by nomisto** [comment]
@matthias-samwald @annabreit I've uploaded the new graph and train/testset to WeTransfer https://we.tl/t-HOolDwPwtW. The password for the zip-Archive is "openbiolink". It seems that the graph contains a little more edges than the previous one, while the train_sample csv contains ca. half the amount of edges than the previous. I don't know if this is an issue or intended.

I created the test/train split with the following parameters:
`--mode rand --test_frac 0.05 --crossval --val 0.05`

**Pull request merged**